cmd/launcher/launcher.go

```diff
@@ -200,51 +205,40 @@ func runLauncher(ctx context.Context, cancel func(), opts *launcher.Options) err
 		checkpointer.SetQuerier(extension)
 	}()
 
-	var controlService *control.ControlService
-
 	// Create the control service and services that depend on it
 	var runner *desktopRunner.DesktopUsersProcessesRunner
-	if opts.ControlServerURL == "" {
+	if k.ControlServerURL() == "" {
 		level.Debug(logger).Log("msg", "control server URL not set, will not create control service")
 	} else {
-		controlService, err = createControlService(ctx, logger, k.ControlStore(), opts)
+		controlService, err := createControlService(ctx, logger, k.ControlStore(), k)
 		if err != nil {
 			return fmt.Errorf("failed to setup control service: %w", err)
 		}
 		runGroup.Add(controlService.ExecuteWithContext(ctx), controlService.Interrupt)
 
-		// serverDataBucketConsumer handles server data table updates
-		controlService.RegisterConsumer(serverDataSubsystemName, k.ServerProvidedDataStore())
-		controlService.RegisterConsumer(agentFlagsSubsystemName, k.AgentFlagsStore())
-
-		desktopEnabledRaw, err := k.AgentFlagsStore().Get([]byte("desktop_enabled_v1"))
-		if err != nil {
-			level.Debug(logger).Log("msg", "failed to query desktop_enabled_v1 flag", "err", err)
-		}
-
-		// For now, while we're in transition from K2 to device trust, we want to default to _not_ running
-		// the desktop process. This can be overridden via control server interaction -- if the desktop_enabled_v1
-		// flag is present (regardless of it's value), this indicates control server has told launcher to enable desktop.
-		desktopProcessSpawningEnabled := desktopEnabledRaw != nil
+		// serverDataConsumer handles server data table updates
+		serverDataConsumer := keyvalueconsumer.New(k.ServerProvidedDataStore())
+		controlService.RegisterConsumer(serverDataSubsystemName, serverDataConsumer)
+		// agentFlagConsumer handles agent flags pushed from the control server
+		agentFlagsConsumer := keyvalueconsumer.New(flagController)
+		controlService.RegisterConsumer(agentFlagsSubsystemName, agentFlagsConsumer)
 
 		runner, err = desktopRunner.New(
+			k,
 			desktopRunner.WithLogger(logger),
 			desktopRunner.WithUpdateInterval(time.Second*5),
 			desktopRunner.WithMenuRefreshInterval(time.Minute*15),
 			desktopRunner.WithHostname(opts.KolideServerURL),
 			desktopRunner.WithAuthToken(ulid.New()),
 			desktopRunner.WithUsersFilesRoot(rootDirectory),
-			desktopRunner.WithProcessSpawningEnabled(desktopProcessSpawningEnabled),
-			desktopRunner.WithGetter(k.AgentFlagsStore()),
+			desktopRunner.WithProcessSpawningEnabled(k.DesktopEnabled()),
```

Comment:
I think, long term, we should replace things like `desktopRunner.WithProcessSpawningEnabled(k.DesktopEnabled())` with having the underlying thing just call `k.DesktopEnabled()` directly.

We don't expect there to be other consumers of `desktopRunner` and the functional arguments exist mostly as a style to manage the developer experience around them. I think knapsack replaces them.

Reply:
Yeah that's a good point, I'll add this to a list of cleanup items. I'll check and see if there are other option style functions that can be eliminated as well.

Reply:
Yeah, it's all the long arc of using knapsack.